File: packages/jsii-pacmak/test/expected.java.jsii-calc-lib/java/org/jsii/tests/calculator/lib/Operation.java
package org.jsii.tests.calculator.lib;
/**
 * Represents an operation on values.
 */
@org.jsii.Jsii(module = org.jsii.tests.calculator.lib.$Module.class, fqn = "@scope/jsii-calc-lib.Operation")
public abstract class Operation extends org.jsii.tests.calculator.lib.Value {
    protected Operation(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    /**
     * String representation of the value.
     */
    public abstract java.lang.String toString();
}

File: packages/jsii-pacmak/test/expected.jsii-calc-lib/dotnet/Amazon.JSII.Tests.Calculator.Lib/Operation.cs
using Amazon.JSII.Runtime.Deputy;

namespace Amazon.JSII.Tests.Calculator.Lib
{
    /// <summary>Represents an operation on values.</summary>
    [JsiiClass(typeof(Operation), "@scope/jsii-calc-lib.Operation", "[]")]
    public abstract class Operation : Value_
    {
        protected Operation(): base(new DeputyProps(new object[]{}))
        {
        }

        protected Operation(ByRefValue reference): base(reference)
        {
        }

        protected Operation(DeputyProps props): base(props)
        {
        }

        /// <summary>String representation of the value.</summary>
        [JsiiMethod("toString", "{\"primitive\":\"string\"}", "[]")]
        public override abstract string ToString();
    }
}